src/main/kotlin/org/radarbase/push/integration/garmin/user/GarminServiceUserRepository.kt

```diff
@@ -50,21 +53,17 @@ class GarminServiceUserRepository(
         clientId = garminConfig.userRepositoryClientId
         clientSecret = garminConfig.userRepositoryClientSecret
 
-        if (tokenUrl != null && clientId.isEmpty())
+        if (clientId.isEmpty())
             throw ConfigException("Client ID for user repository is not set.")
 
-        repositoryClient = if (tokenUrl != null) {
-            OAuth2Client.Builder()
+        repositoryClient = OAuth2Client.Builder()
                 .credentials(clientId, clientSecret)
                 .endpoint(tokenUrl)
                 .scopes("SUBJECT.READ", "MEASUREMENT.READ", "SUBJECT.UPDATE")
                 .httpClient(client)
                 .build()
-        } else OAuth2Client.Builder().build()
 
-        basicCredentials = if (tokenUrl == null && clientId != null) {
-            Credentials.basic(clientId, clientSecret)
-        } else null
+        basicCredentials = if (clientId.isNotEmpty()) Credentials.basic(clientId, clientSecret) else null
```

Comment:
If you already know that `tokenUrl` is not null and `clientId` is not null, you don't need `basicCredentials` at all. `requestAuthorization` can then be simplified to only include the `repositoryClient` flow. It can then return a non-null String, and the null-check in `requestFor` is then not needed.